File: easy/simple_or_trump.rb
V = { '2' => 2, '3' => 3, '4' => 4, '5' => 5, '6' => 6, '7' => 7, '8' => 8,
      '9' => 9, '1' => 10, 'J' => 11, 'Q' => 12, 'K' => 13, 'A' => 14 }.freeze
def rank(c, t)
  r = V[c[0]]
  r += 13 if c[-1] == t
  r
end

def win(l, r, t)
  case rank(l, t) <=> rank(r, t)
  when -1 then r
  when 1 then l
  else l + ' ' + r
  end
end

File.open(ARGV[0]).each_line do |line|
  s = line.split
  puts win(s[0], s[1], s[3])
end

File: easy/simple_or_trump.lua
v = {10, 2, 3, 4, 5, 6, 7, 8, 9, J = 11, Q = 12, K = 13, A = 14}
function rank (a, t)
  local r
  if tonumber(a:sub(1, 1)) ~= nil then
    r = v[tonumber(a:sub(1, 1))]
  else
    r = v[a:sub(1, 1)]
  end
  if a:sub(#a, #a) == t then
    r = r + 13
  end
  return r
end

function win (l, r, t)
  local lr, rr = rank(l, t), rank(r, t)
  if lr > rr then
    return l
  elseif lr < rr then
    return r
  end
  return l .. " " .. r
end

for line in io.lines(arg[1]) do
  local l, r, t = line:match("(%w+) (%w+) | (%u)")
  print(win(l, r, t))
end
